api/agent.go

```diff
@@ -116,7 +116,12 @@ func (a *api) createLoadAgentHandler(w http.ResponseWriter, r *http.Request) {
 		})
 		return
 	}
-	a.setResource(agentId, lt)
+	if ok := a.setResource(agentId, lt); !ok {
+		writeAgentResponse(w, http.StatusBadRequest, &client.AgentResponse{
```

Comment:
Can we return a 409 conflict to make us look more classy? :)

Reply:
I am always in for some class :+1: